### PR TITLE
Add API benchmarks

### PR DESCRIPTION
**Pull request by VoVAllen**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: benchmarks/benchmarks/api/bench_edge_ids.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import numpy as np
+
+from .. import utils
+
+# edge_ids is not supported on cuda
+@utils.skip_if_gpu()
+@utils.benchmark('time', timeout=1200)
+@utils.parametrize_cpu('graph_name', ['cora', 'livejournal', 'friendster'])
+@utils.parametrize_gpu('graph_name', ['cora', 'livejournal'])
+@utils.parametrize('format', ['csc'])  # csc is not supported
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_edge_ids.py:13]: csr is not supported

File: benchmarks/benchmarks/api/bench_edge_ids.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@utils.parametrize('fraction', [0.01, 0.1])
+def track_time(graph_name, format, fraction):
+    device = utils.get_bench_device()
+    graph = utils.get_graph(graph_name, format)
+    graph = graph.to(device)
+    eids = np.random.RandomState(6666).choice(
+        np.arange(graph.num_edges(), dtype=np.int64), int(graph.num_edges()*fraction))
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_edge_ids.py:20]: We have set the seed at here: https://github.com/dmlc/dgl/blob/master/benchmarks/benchmarks/utils.py#L251

File: benchmarks/benchmarks/api/bench_edge_ids.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import time
+import dgl
+import torch
+import numpy as np
+
+from .. import utils
+
+# edge_ids is not supported on cuda
+@utils.skip_if_gpu()
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_edge_ids.py:9]: should not skip

**Review by jermainewang — APPROVED** [review]
Mostly good. Some comments.

**Comment by jermainewang** [comment]
cc @zheng-da here. For those API tests, I feel only having raw time measurement is not very clear for whether the performance has reached the peak performance. Shall we consider using some throughput number here?

File: benchmarks/benchmarks/api/bench_edge_ids.py (hunk 3 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_edge_ids.py:9]: Edge id is not supported on gpu

Comment (jermainewang) [reply benchmarks/benchmarks/api/bench_edge_ids.py:9]: The following works for me:
```python
import dgl
import torch

g1 = dgl.graph(([0, 1], [1, 2])).to('cuda:0')

print(g1.edge_ids([0, 1], [1, 2]))
```

File: benchmarks/benchmarks/api/bench_edge_ids.py (hunk 2 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_edge_ids.py:21]: Removed RandomState

File: benchmarks/benchmarks/api/bench_edge_ids.py (hunk 3 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_edge_ids.py]: Fixed
